Neuro/Minigames/Solvers/CleanO2FilterSolver.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections;
+using UnityEngine;
+using Neuro.Cursor;
+using System.Linq;
+
+namespace Neuro.Minigames.Solvers;
+
+[MinigameSolver(typeof(LeafMinigame))]
+public class CleanO2FilterSolver : MinigameSolver<LeafMinigame>
+{
+    protected override IEnumerator CompleteMinigame(LeafMinigame minigame, NormalPlayerTask task)
+    {
+        // ValidArea is somewhere offscreen so easier to just move towards the arrows on the chute
+        Vector3 exit = minigame.Arrows[0].transform.position;
+        Collider2D lastLeaf = null;
+        while (!task.IsComplete)
+        {
+            Collider2D _lastLeafCopy = lastLeaf;
+            Collider2D leaf = minigame.Leaves
+                .Where(l => l && l.enabled && l.gameObject.active && l != _lastLeafCopy
+                    && Vector3.Distance(l.transform.position, exit) > 4f)  
+                .MaxBy(l => (InGameCursor.Instance.Position - (Vector2)l.transform.position).magnitude);
+
+            if (!leaf)
+            {
+                yield return null;
+                continue;
+            }
```

Comment:
If there is only one leaf left, we shouldn't ignore it... right?
```suggestion
            if (!leaf) leaf = lastLeaf;
            
            if (!leaf)
            {
                yield return null;
                continue;
            }
```

Reply:
I believe this will run into the same issue as above, as there is a chance that the cursor will grab the object while it is fully sucked in and being deleted, causing an error. From my testing once the leaf is past the `4f` distance point it will get sucked in automatically without the need for interaction.